packages/menu/src/MenuItem/MenuItem.stories.tsx

```diff
@@ -1,49 +1,173 @@
 /* eslint-disable react/jsx-key */
 /* eslint-disable react/display-name */
 import React from 'react';
-import { StoryMetaType } from '@lg-tools/storybook-utils';
+import { InstanceDecorator, StoryMetaType } from '@lg-tools/storybook-utils';
+import { StoryObj } from '@storybook/react';
 
-import Icon from '@leafygreen-ui/icon';
+import { css } from '@leafygreen-ui/emotion';
+import Icon, { glyphs } from '@leafygreen-ui/icon';
 import { Theme } from '@leafygreen-ui/lib';
 
-import { MenuProps } from '../Menu/Menu.types';
+import { Menu, MenuProps } from '../Menu';
 import { MenuContext } from '../MenuContext';
 import { Size } from '../types';
 
-import MenuItem from './MenuItem';
+import { MenuItem, Variant } from '.';
+
+/** Implements a MenuContext wrapper around each `MenuItem` */
+const _withMenuContext =
+  (): InstanceDecorator<typeof MenuItem & typeof Menu> => (Instance, ctx) => {
+    const {
+      args: { darkMode: darkModeProp, renderDarkMenu, highlighted, ...props },
+    } = ctx ?? {
+      args: {
+        darkMode: false,
+        renderDarkMenu: false,
+        highlighted: false,
+      },
+    };
+
+    const darkMode = (renderDarkMenu || darkModeProp) ?? false;
+    const theme = darkMode ? Theme.Dark : Theme.Light;
+
+    return (
+      <MenuContext.Provider
+        value={{
+          highlightIndex: highlighted ? -1 : undefined,
+          darkMode,
+          theme,
+        }}
+      >
+        <ul
+          className={css`
+            width: 256px;
+          `}
+        >
+          <MenuItem {...props} />
+        </ul>
+      </MenuContext.Provider>
+    );
+  };
 
 export default {
   title: 'Components/Menu/MenuItem',
   component: MenuItem,
+  args: {
+    children: 'Menu Item',
+    active: false,
+  },
   parameters: {
     default: null,
     generate: {
+      storyNames: [
+        'Default',
+        'Active',
+        'Focused',
+        'Destructive',
+        'Disabled',
+        'DarkInLightMode',
+      ],
       combineArgs: {
         darkMode: [false, true],
         description: [undefined, 'This is a description'],
-        glyph: [undefined, <Icon glyph="Cloud" size="large" />],
-        active: [false, true],
-        size: Object.values(Size),
+        glyph: [undefined, <Icon glyph="Cloud" />],
+        size: [Size.Default, Size.Large],
+      },
+      decorator: _withMenuContext(),
+    },
+  },
+} satisfies StoryMetaType<typeof MenuItem, Partial<MenuProps>>;
+
+export const LiveExample = {
+  args: {
+    as: 'button',
+    description: 'Description',
+    glyph: undefined,
+  },
+  argTypes: {
+    description: { control: 'text' },
+    glyph: {
+      control: 'select',
+      options: [undefined, ...Object.keys(glyphs)],
+    },
+    size: {
+      control: 'select',
+      options: Object.values(Size),
+    },
+    renderDarkMenu: {
+      control: 'boolean',
+    },
+  },
+  render: ({ children, glyph, ...args }) => (
+    // @ts-expect-error
```

Comment:
is this getting removed in a later PR? if not, is it worth adding a note on what this is added for?